fRanz/src/producer.cpp

```diff
@@ -32,20 +32,33 @@ SEXP GetRdProducer(Rcpp::StringVector keys, Rcpp::StringVector values) {
 int KafkaProduce(SEXP producer_pointer,
                  SEXP topic, 
                  Rcpp::IntegerVector partition, 
-                 SEXP key, 
-                 SEXP value) {
+                 Rcpp::StringVector keys, 
+                 Rcpp::StringVector values) {
     Rcpp::XPtr<RdKafka::Producer> producer(producer_pointer);
     std::string s_topic = Rcpp::as<std::string>(topic);
-    std::string s_value = Rcpp::as<std::string>(value);
-    std::string s_key = Rcpp::as<std::string>(key);
+
+    if (keys.size() != values.size()) {
+        std::cout << "keys and values must be same size" << std::endl;
+        return -1;
+    }
+    int numMsgs = keys.size();
+    int numSent = 0;
+
+    for (int i = 0; i < numMsgs; i++) {
+        std::string s_value = Rcpp::as<std::string>(values[i]);
+        std::string s_key = Rcpp::as<std::string>(keys[i]);
     
-    RdKafka::ErrorCode resp = producer->produce(
-        s_topic, partition[0],  RdKafka::Producer::RK_MSG_COPY,
-        const_cast<char *>(s_value.c_str()),s_value.size(),
-        const_cast<char *>(s_key.c_str()), s_key.size(),
-        0, NULL
-    );
+        RdKafka::ErrorCode resp = producer->produce(
+            s_topic, partition[0],  RdKafka::Producer::RK_MSG_COPY,
+            const_cast<char *>(s_value.c_str()),s_value.size(),
+            const_cast<char *>(s_key.c_str()), s_key.size(),
+            0, NULL
+        );
+
+        if (resp == RdKafka::ERR_NO_ERROR) numSent++;
```

Comment:
We really need to figure out if we do best effort or not. Maybe we return 'non sent messages' or something